RIOT-OS/RIOT
# examples/leds_shell: add example for interactive LEDs/GPIO control

### Contribution description

**Rationale:** When I remember my first steps as RIOT newbie, I would like as soon as possible flash the program and
show that it works and could control something, for example LED. However, for RIOT newbie this is not so easy. Of course there is `examples/blink` - but it works automatically. Manual control is something, in my humble opinion, missing.

This simple program allows easy control of internal LEDs via RIOT CLI. Additionally, it allows initialization of any GPIO port in output mode and manually set HIGH/LOW state.

From the programming learning perspective, analysis and experiments with this program could teach:
- RIOT CLI,
- Adding own functions,
- Basic GPIO.

Any comments are welcome. 

### Testing procedure

Flash `examples/LEDs` and check if number of internal LEDs is properly detected.

I tested it with boards from 1 to 4 LEDs, respectively: `nucleo-l496rg`, `stm32l47g-disco/native', 'nucleo-f207zg` and  `stm32f469i-disco`.

Read README.md and check if everything is clear.

### Issues/PRs references

PR #20637
